src/Core/Domain/Product/Image/Command/UpdateImagesCommand.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Image\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Image\DTO\ImageCollection;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+
+/**
+ * Updates product images.
+ */
+class UpdateImagesCommand
```

Comment:
How about `UpdateProductImagesCommand`?

I'm also started to think what is image update? Shouldn't we have "upload images" instead? If so, maybe it's not Domain responsibility to do so in the first place?

Reply:
you mean it is not the domain responsibility to upload any kind of file or image? It also assigns image id with product id.

Reply:
Assigning image to product is part of domain, but dealing with filesystem is not, although it's done in other handlers as far as I remember.